src/mono/msbuild/AotCompilerTask/MonoAotCompiler.cs

```diff
@@ -0,0 +1,249 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Concurrent;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+public class MonoAotCompiler : Microsoft.Build.Utilities.Task
+{
+    /// <summary>
+    /// Path to AOT cross-compiler binary (mono-aot-cross)
+    /// </summary>
+    [Required]
+    public string AotCompilerBinary { get; set; } = ""!;
```

Comment:
I think Aot prefix is redundant here, it should include Path in name if it's about path to binary